app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -1435,13 +1435,26 @@ class BrowserTabViewModel @Inject constructor(
         viewModelScope.launch { updateBookmarkAndFavoriteState(url) }
     }
 
+    private fun stripBasicAuthFromUrl(url: String): String {
+        val uri = URI(url)
+        val userInfo = uri.userInfo
+
+        if (userInfo != null) {
+            val queryStr = uri.rawQuery?.let { "?$it" } ?: ""
+            val uriFragment = uri.fragment?.let { "#$it" } ?: ""
+            return "${uri.scheme}://${uri.host}${uri.path}$queryStr$uriFragment"
+        }
+
+        return url
+    }
```

Comment:
This change affects the URL bar so I'd expect a very comprehensive set of unit tests (not only them maestro ones) to make sure we're covering decent amount of cases

Reply:
Thanks @aitorvs - sounds good to me, I'll get those unit tests written up 👍🏼